packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/tree/ViewTreeDescriptionConverter.java

```diff
@@ -222,4 +229,26 @@ private StyledString getTreeItemLabel(AQLInterpreter interpreter, VariableManage
         }
         return result;
     }
+
+    private List<String> getTreeRepresentationIcons(AQLInterpreter interpreter, VariableManager variableManager, String iconExpression) {
+        List<String> result = List.of();
+
+        if (iconExpression.isBlank()) {
+            result = this.representationImageProviders.stream()
```

Comment:
Same comment (you really can create a dedicated class instead of copying and pasting the same code everywhere)